test/provider_test.exs

```diff
@@ -48,7 +48,11 @@ defmodule ProviderTest do
       System.put_env("OPT_6", "false")
       System.put_env("OPT_7", "3.14")
 
+      {:ok, pid} = TestModule.start_link()
```

Comment:
Do this instead

```elixir
pid = start_supervised!(TestModule, restart: :temporary)
```

And you won't need to manually stop the server. As [the docs](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#start_supervised/2) say:

> The advantage of starting a process under the test supervisor is that it is guaranteed to exit before the next test starts.